packages/perspective/transaction/classify.ts

```diff
@@ -7,14 +7,15 @@ export const classifyTransaction = (txv?: TransactionView): TransactionClassific
     return 'unknown';
   }
 
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'swap')) return 'swap';
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'swapClaim')) return 'swapClaim';
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'delegate')) return 'delegate';
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'undelegate')) return 'undelegate';
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'undelegateClaim'))
-    return 'undelegateClaim';
-  if (txv.bodyView?.actionViews.some(a => a.actionView.case === 'ics20Withdrawal'))
-    return 'ics20Withdrawal';
+  const allActionCases = new Set(txv.bodyView?.actionViews.map(a => a.actionView.case));
```

Comment:
Refactored this to use a `Set` so we're not iterating over the action views so many times like we were previously.